project/vision_backend/management/commands/vb_export_spacer_data.py

```diff
@@ -43,43 +55,58 @@ def source_meta_json(source):
         return json.dumps(source.to_dict(), indent=2)
 
     @staticmethod
-    def image_meta_json(metadata):
-        return json.dumps(metadata.to_dict(), indent=2)
+    def image_meta_json(image):
+        # Read out default image metadata.
+        metadata_dict = image.metadata.to_dict()
+        # Add details re. train vs valset.
+        metadata_dict['in_trainset'] = image.trainset
+        metadata_dict['in_valset'] = image.valset
+        return json.dumps(metadata_dict, indent=2)
 
     @staticmethod
     def image_annotations_json(image):
         rowcols = []
-        for ann in Annotation.objects.filter(image=image).\
+        for ann in Annotation.objects.filter(image=image). \
+            order_by('point__id'). \
                 annotate(row=F('point__row'),
                          col=F('point__column'),
                          label_id=F('label__pk')):
-            rowcols.append({'row': ann.row, 'col': ann.col, 'label': ann.label_id})
+            rowcols.append({'row': ann.row,
+                            'col': ann.col,
+                            'label': ann.label_id})
         return json.dumps(rowcols, indent=2)
 
     def handle(self, *args, **options):
 
         # Start by exporting the label-set
-        c = boto.connect_s3(settings.AWS_ACCESS_KEY_ID, settings.AWS_SECRET_ACCESS_KEY)
+        c = boto.connect_s3(settings.AWS_ACCESS_KEY_ID,
+                            settings.AWS_SECRET_ACCESS_KEY)
         bucket = c.get_bucket(options['bucket'])
-        label_set_key = Key(bucket, name=options['name']+'/' + 'label_set.json')
+        label_set_key = Key(bucket,
+                            name=options['name']+'/' + 'label_set.json')
         label_set_key.set_contents_from_string(self.labelset_json)
 
         # Filter sources
         sources = Source.objects.filter()
         sources = filter_out_test_sources(sources)
         sources = [s for s in sources if
                    s.nbr_confirmed_images > options['min_required_imgs']]
-        sources = sources[options['skip_to']:]
 
         # Iterate over sources
         for itt, source in enumerate(sources):
-            print("Exporting source id:{}. [{}({})] with {} images...".format(source.pk, itt+1, len(sources),
-                                                                              source.nbr_confirmed_images))
+            print("Exporting {}, id:{}. [{}({})] with {} images...".format(
```

Comment:
Add a `u` before the opening quote to make it a Unicode string, so that those non-ASCII source names don't trip you up. `print(u"Exporting {}, id:{}. [{}({})] with {} images...".format(`

Reply:
Thanks! Changed.